### PR TITLE
EE image variables

### PR DESCRIPTION
**Pull request by craig-br**
##### SUMMARY
**EE image and registry variables**
- This will make it easier to change the EE registry and versions
  -  `ee_registry_name: ` EE image repo
  -  `ee_images:` - List of controller EEs to pull for the workshop
  -  `ee_default_image:` - Default EE image name used in controller. This EE image has custom credentials defined during controller configuration.

**30_controller.yml**
- Updated to use new variables

**README.md**
- Added variable examples for `extra_vars.yml` file
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
- provisioner

### CODE REVIEW

File: provisioner/README.md [hunk 1]
```diff
@@ -94,6 +94,28 @@ ec2_xtra:
     username: ec2-user
     os_type: linux
     size: r5b.2xlarge
+
+# Registry name to download execution environments
```
Comment (IPvSean) [inline provisioner/README.md:98]: Why are we adding this giant block to the README.md?

File: roles/control_node/defaults/main.yml [hunk 2]
```diff
@@ -9,3 +9,16 @@ network_ee: "registry.redhat.io/ansible-automation-platform-20-early-access/ee-s
...
 windows_ee: "quay.io/acme_corp/windows_ee:latest"
 smart_mgmt_ee: "quay.io/s4v0/ee-automated-smart-mgmt-29:1.0.4"
+
+
+# EE registry name used in installer and EE controller credential
+ee_registry_name: registry.redhat.io
+
+# EEs to download for awx user
+ee_images:
+  - "{{ ee_registry_name }}/ansible-automation-platform-20-early-access/ee-supported-rhel8:2.0.1"
```
Comment (IPvSean) [inline roles/control_node/defaults/main.yml:19]: why not point to latest?  are we worried they will be buggy?

**Review by IPvSean — COMMENTED** [review]
looks really good outside 1 comment on the README (I don't think we need to add that to the readme? just add like the ee_registry_name and an example....

I had one question on pointing to latest

File: roles/control_node/defaults/main.yml (hunk 2 above)
Comment (craig-br) [reply roles/control_node/defaults/main.yml:19]: I remember we had issues with using `:latest` tag for EEs. It always tried to fetch the latest version before it executes the job template. Is that still accurate? If not, I'll change it to `:latest`

File: provisioner/README.md [hunk 3]
```diff
@@ -94,6 +94,20 @@ ec2_xtra:
...
+ee_registry_name: registry.redhat.io
+
+# List of execution environments to download during controller installation:
+ee_images:
+   - "{{ ee_registry_name }}/ansible-automation-platform-20-early-access/ee-supported-rhel8:2.0.1"
+   - "{{ ee_registry_name }}/ansible-automation-platform-20-early-access/ee-29-rhel8:2.0.1"
+   - "{{ ee_registry_name }}/ansible-automation-platform-20-early-access/ee-minimal-rhel8:2.0.1"
```
Comment (rlopez133) [inline provisioner/README.md:105]: AAP 2.1 controller installation should auto install the proper images. Also these images are early access not latest GA.

Comment (craig-br) [reply provisioner/README.md:105]: @rlopez133 . Agreed it's the incorrect version. What about the scenario where the new versions of the EEs are released asynchronous to AAP version. I believe having these vars will accommodate that

Comment (rlopez133) [reply provisioner/README.md:105]: @craig-br true, they likely just need to be optional.

Comment (craig-br) [reply provisioner/README.md]: @rlopez133 I've updated the EE images to AAP 21. The vars I'm proposing are all optional. The examples are specified under the `Optional` section of the `extra_vars.yml` README

**Pull request merged**